crates/vm/src/arch/integration_api.rs

```diff
@@ -370,6 +370,10 @@ where
     fn num_public_values(&self) -> usize {
         self.core.num_public_values()
     }
+
+    fn columns(&self) -> Vec<String> {
+        self.core.columns()
+    }
```

Comment:
I understand that it should also return the columns of the adapter, but as it does not implement `BaseAirWithPublicValues`, it is not possible (or at least not trivially). Maybe I should also implement columns for (directly and only in) `BaseAir`, but it's an external struct, that's why I discarded it at first. I'll take a deeper look while we solve the rest of the questions.